store/mockstore/mocktikv/mppserver.go

```diff
@@ -0,0 +1,225 @@
+package mocktikv
+
+import (
+	"context"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/kvproto/pkg/coprocessor"
+	"github.com/pingcap/kvproto/pkg/kvrpcpb"
+	"github.com/pingcap/kvproto/pkg/tikvpb"
+	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+)
+
+// CreateRPCServer creates a rpc server for mpp.
+func CreateRPCServer() *grpc.Server {
+	defer func() {
+		if v := recover(); v != nil {
+			logutil.BgLogger().Error("panic in mpp server", zap.Any("stack", v))
+		}
+	}()
+
+	s := grpc.NewServer()
+	tikvpb.RegisterTikvServer(s, &mppServer{})
+	return s
+}
+
+type mppServer struct {
+	MockGrpcServer
+}
+
+// Coprocessor implements the TiKVServer interface.
+func (c *mppServer) Coprocessor(ctx context.Context, in *coprocessor.Request) (*coprocessor.Response, error) {
+	defer func() {
+		if v := recover(); v != nil {
+			logutil.BgLogger().Error("panic in mpp server coprocessor", zap.Any("stack", v))
+		}
+	}()
+	handler := &rpcHandler{}
+	res := handler.handleCopDAGRequest(in)
+	return res, nil
+}
+
+// MockGrpcServer mock a TiKV gprc server for testing.
+type MockGrpcServer struct{}
```

Comment:
Compound the `TiKVServer` interface and remove the redundant empty methods.

Reply:
We still need a `MockGrpcServer `, I just move this from the test file. https://github.com/pingcap/tidb/pull/13065/files#diff-68587a721b9d0df9b704d47f646e3a3eL208

Reply:
Reuse the `TiKVServer`  interface is to reuse the client logic. otherwise, we should use different grpcClient to send difference requests.

Reply:
Oh, I got you, Great idea.